### PR TITLE
Add gcs public bucket for prow staging logs.

### PR DESCRIPTION
**Pull request by ameukam**
Following prow [documentation](https://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#configure-a-gcs-buckethttps://github.com/kubernetes/test-infra/blob/master/prow/getting_started_deploy.md#configure-a-gcs-bucket) guidance :
Create a gcs bucket for tide history and build logs.
Create a service account and grant admin access to the bucket.
Create a service account key and add the generated key to Secret Manager.

Signed-off-by: Arnaud Meukam <ameukam@gmail.com>

### CODE REVIEW

**Comment by ameukam** [comment]
The service account key generated should be added as a secret in namespace `prow`.

File: infra/gcp/ensure-main-project.sh [hunk 1]
```diff
@@ -84,6 +89,45 @@ enable_api "${PROJECT}" secretmanager.googleapis.com
 color 6 "Ensuring the cluster terraform-state bucket exists"
 ensure_private_gcs_bucket "${PROJECT}" "gs://${CLUSTER_TERRAFORM_BUCKET}"
 
+
+color 6 "Ensuring all the prow buckets"
```
Comment (MushuEE) [inline infra/gcp/ensure-main-project.sh:93]:
```suggestion
color 6 "Ensuring all the prow buckets exist"
```
Super Nit just to keep it consistent with the rest of the language

Comment (ameukam) [reply infra/gcp/ensure-main-project.sh]: Thank you! :-)

File: infra/gcp/ensure-main-project.sh [hunk 2]
```diff
@@ -57,6 +57,11 @@ CLUSTER_USERS_GROUP="gke-security-groups@kubernetes.io"
 # The DNS admins group.
 DNS_GROUP="k8s-infra-dns-admins@kubernetes.io"
 
+# Buckets for the logs of prow
+PROW_BUCKETS=(
+    k8s-prow-staging-logs
+)
```
Comment (spiffxp) [inline infra/gcp/ensure-main-project.sh:63]: If the intent is to use this like gs://kubernetes-jenkins, I feel like we're going to want those sorts of buckets not in kubernetes-public.  It'll be easier to keep track of CI artifact costs if they're in their own project.  I'm open to suggestions.

File: infra/gcp/ensure-main-project.sh [hunk 3]
```diff
@@ -84,6 +89,45 @@ enable_api "${PROJECT}" secretmanager.googleapis.com
...
+
+    color 6 "Empowering service account: ${SERVICE_ACCOUNT_NAME}"
+    empower_svcacct_to_write_gcs_bucket "${SERVICE_ACCOUNT_EMAIL}" "gs://${bucket}"
+
+    color 6 "Creating private key for service account: ${SERVICE_ACCOUNT_NAME}"
+    gcloud iam service-accounts keys create "${KEY_FILE}" \
+        --project "${PROJECT}" \
+        --iam-account "${SERVICE_ACCOUNT_EMAIL}"
+
+    color 6 "Creating secret to store private key"
```
Comment (spiffxp) [inline infra/gcp/ensure-main-project.sh:119]: Who can access the secret? I would expect to see a `gcloud secrets add-iam-policy-binding` call

My suggestion would be k8s-infra-prow-oncall@ (I'm happy to approve a PR adding you as a member to that group)

**Review by spiffxp — COMMENTED** [review]
Same comment as https://github.com/kubernetes/k8s.io/issues/1475#issuecomment-742660227 - I would like to better understand what the plan is

File: infra/gcp/ensure-main-project.sh (hunk 2 above)
Comment (ameukam) [reply infra/gcp/ensure-main-project.sh:63]: The PR is more in favor of the support multiple buckets by tide. I didn't consider the billing aspect.

File: infra/gcp/ensure-main-project.sh (hunk 3 above)
Comment (ameukam) [reply infra/gcp/ensure-main-project.sh:119]: I'll push the binding once the right option is chosen for `gs://kubernetes-jenkins` migration.

**Comment by ameukam** [comment]
🤔 Force push on this branch doesn't update the PR. 2 things changed: 
- switch main as default branch
- pull-k8sio-yamllint has been replaced by pull-k8s-verify.
Will open a new PR tomorrow.

**Comment by ameukam** [comment]
/close
For #1909

**Pull request closed without merging**